# base callmkdir /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/guppy_fastq/home/yuxin/ont-guppy-cpu/bin/guppy_basecaller \-i /data/yuxin/2021Fall/HomoSapiens/HMEC/HMEC_WT/fast5_single_pass \-s /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/guppy_fastq \--flowcell FLO-MIN107 --kit SQK-RNA001 \--recursive \--num_callers 12# tombotombo resquiggle /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/fast5/pass \/data/yuxin/2021Fall/Reference/HomoSapiens/hg38_bychr.fa \--processes 4 --num-most-common-errors 5 --overwritemkdir /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/tombo_resulttombo detect_modifications de_novo \--fast5-basedirs /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/fast5 \--statistics-file-basename /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/tombo_result/de_novo_detection \--processes 5tombo text_output browser_files \--fast5-basedirs /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/fast5 \--statistics-filename /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/tombo_result/de_novo_detection.tombo.stats \--browser-file-basename /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/tombo_result/de_novo_detection \--file-types dampened_fraction coverage# MINESwig2bed </data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/tombo_result/de_novo_detection.dampened_fraction_modified_reads.minus.wig> \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/tombo_result/de_novo_detection.dampened_fraction_modified_reads.minus.wig.bedwig2bed </data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/tombo_result/de_novo_detection.dampened_fraction_modified_reads.plus.wig> \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/tombo_result/de_novo_detection.dampened_fraction_modified_reads.plus.wig.bedmkdir /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/mines_resultpython /home/yuxin/MINES/genomic_MINES.py \--fraction_modified_plus /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/tombo_result/de_novo_detection.dampened_fraction_modified_reads.plus.wig.bed \--fraction_modified_minus /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/tombo_result/de_novo_detection.dampened_fraction_modified_reads.minus.wig.bed \--coverage_plus /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/tombo_result/de_novo_detection.coverage.plus.bedgraph \--coverage_minus /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/tombo_result/de_novo_detection.coverage.minus.bedgraph \--output /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/mines_result/mines_result.bed \--ref /data/yuxin/2021Fall/Reference/HomoSapiens/hg38.fa# mapcd /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/fastq_guppyfind -maxdepth 5 -name "*.fastq" > /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/fastq_list.txttar -T /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/fastq_list.txt \-czvf /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/fastq.tar.gzmkdir /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam# map to genomic reference, done/home/yuxin/minimap2-2.17_x64-linux/minimap2 -ax splice \/data/yuxin/2021Fall/Reference/HomoSapiens/hg38.fa \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/fastq.tar.gz > \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/g.sam -t 6samtools view -b \-S /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/g.sam > \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/g.bamsamtools sort \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/g.bam \-o /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/g.bamsamtools index \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/g.bam \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/g.bam.bai# map to transcriptome/home/yuxin/minimap2-2.17_x64-linux/minimap2 -ax splice \/data/yuxin/2021Fall/Reference/HomoSapiens/gencode.v29.transcripts.fa \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/fastq.tar.gz > \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/t.sam -t 6samtools view -b \-S /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/t.sam > \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/t.bamsamtools sort \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/t.bam \-o /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/t.bamsamtools index \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/t.bam \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/t.bam.bai# DRUMMERbedtools bamtobed -i /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_shMETTL3/bam/t.bam \| cut -f1 > /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_shMETTL3/bam/t.txtbedtools bamtobed -i /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/t.bam \| cut -f1 > /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/t.txtmkdir /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/drummer_resultsort /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/t.txt \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_shMETTL3/bam/t.txt \| uniq -d > /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/drummer_result/transcripts.txtpython /home/yuxin/DRUMMER/DRUMMER.py \-r /data/yuxin/2021Fall/Reference/HomoSapiens/gencode.v29.transcripts_simplified.fa \-l /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/drummer_result/transcripts.txt \-o /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/drummer_result \-c /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/t.bam \-t /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_shMETTL3/bam/t.bam -a isoform# ELIGOSmkdir /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/eligos_resulteligos2 pair_diff_mod \-tbam /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/g.bam \-cbam /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/g.bam \-reg /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/g.bed \-ref /data/yuxin/2021Fall/Reference/HomoSapiens/hg38.fa \-o /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/eligos_result_diffmod \-t 6# differrpython3 /home/yuxin/differr_nanopore_DRS/differr/main.py \-a /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_shMETTL3/bam/g.bam \-b /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/g.bam \-r /data/yuxin/2021Fall/Reference/HomoSapiens/hg38.fa \-o /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/differr_result/differr.bed# nanopolish## nanopolish, wtmkdir /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/xpore_resultcd /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/fastq_guppyfind -maxdepth 5 -name "*.fastq"|xargs -i awk \'{if(NR%4 == 1){print ">" substr($0, 2)}}{if(NR%4 == 2){print}}' {} |cat > \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/xpore_result/reads.fasta## index/home/yuxin/nanopolish/nanopolish index -d /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/fast5/pass \/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/xpore_result/reads.fasta/home/yuxin/nanopolish/nanopolish eventalign \--reads /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/xpore_result/reads.fasta \--bam /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/bam/t.bam \--genome /data/yuxin/2021Fall/Reference/HomoSapiens/gencode.v29.transcripts_simplified.fa \--scale-events --signal-index \--summary /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/xpore_result/summary.txt \> /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/xpore_result/reads-ref.eventalign.txt \-t 10# xporexpore dataprep \--eventalign /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/xpore_result/reads-ref.eventalign.txt \--out_dir /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/xpore_result \--n_processes 10xpore diffmod --config=/data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/xpore_result/hek293_config.yml \--n_processes 10# m6Anetmkdir /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/m6Anet_resultm6anet-dataprep \--eventalign /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/xpore_result/reads-ref.eventalign.txt \--out_dir /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/m6Anet_result \--n_processes 10m6anet-run_inference --input_dir /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/m6Anet_result \--out_dir /data/yuxin/2021Fall/HomoSapiens/HEK293/HEK_WT/m6Anet_result \--n_processes 4